client/src/components/CustomFields.js

```diff
@@ -659,10 +659,14 @@ function getInvisibleFields(
 export const CustomFieldsContainer = props => {
   const { parentFieldName, formikProps } = props
   const [invisibleFields, setInvisibleFields] = useState([])
+  const prevInvisibleFields = useRef(invisibleFields)
```

Comment:
I am not sure but i feel like doing previous value checking suggests me that we are maybe not initializing the state correctly. If doing nothing updates the state, then we should probably initialize the `invisibleFields` with `null` or something which doesn't cause a rerender. That would be a simpler solution

Reply:
The problems is that React hooks do shallow (object) equality, and if the object is re-created (but with the exact same properties), it *will* fire the hook. The `_isEqual()` comparing to the previous value is an optimisation/work-around for that.